my_enumerables.rb

```diff
@@ -0,0 +1,190 @@
+# rubocop:disable Metrics/ModuleLength
+# rubocop:disable Metrics/CyclomaticComplexity
+# rubocop:disable Metrics/PerceivedComplexity
+module Enumerable
+  def my_each(pro = nil)
+    return to_enum unless block_given?
+
+    arr = to_a
+    i = 0
+
+    if pro
+      while i < arr.length
+        proc.cal arr[i]
+        i += 1
+      end
+    else
+      while i < arr.length
+        yield arr[i]
+        i += 1
+      end
+    end
+    arr
+  end
+
+  def my_each_with_index(pro = nil)
+    return to_enum unless block_given?
+
+    arr = to_a
+    i = 0
+
+    if pro
+      while i < arr.length
+        proc.cal arr[i], i
+        i += 1
+      end
+    else
+      while i < arr.length
+        yield arr[i], i
+        i += 1
+      end
+    end
+
+    arr
+  end
+
+  def my_select
+    return to_enum unless block_given?
+
+    arr = to_a
+    arr2 = []
+    arr.my_each_with_index do |_item, index|
+      arr2.push(arr[index]) if yield arr[index]
+    end
+    arr2
+  end
+
+  def my_all?(args = nil)
+    return to_enum unless block_given? || !args.nil?
+
+    arr = to_a
+    if block_given?
+      arr.my_each_with_index do |_item, index|
+        return false unless yield arr[index]
+      end
+    elsif args.is_a? Class
+      arr.my_each_with_index do |item, _index|
+        return false unless item.class.ancestors.include?(args)
+      end
+    elsif args.is_a? Regexp
+      arr.my_each_with_index do |item, _index|
+        return false unless item.match(args)
+      end
+    end
+    true
+  end
```

Comment:
#my_all? when no block or argument is given returns true when none of the collection members are false or nil
#my_all? when no block or argument is given returns false when one of the collection members are false or nil
#my_all? when a pattern other than Regex or a Class is given returns false if any of the collection does not match the pattern